server/main.py

```diff
@@ -66,7 +71,10 @@ async def upsert(
     token: HTTPAuthorizationCredentials = Depends(validate_token),
 ):
     try:
-        ids = await datastore.upsert(request.documents)
+        documents = request.documents
+        if request.preprocess_options and request.preprocess_options.summarize:
```

Comment:
I believe you meant `summary`

Reply:
Thanks for the catch! I will wait for @isafulf 's review and address feedback in a separate commit